code/game/objects/items/mail.dm

```diff
@@ -332,3 +334,166 @@
 /obj/item/paper/fluff/junkmail_generic/Initialize(mapload)
 	default_raw_text = pick(GLOB.junkmail_messages)
 	return ..()
+
+// =====================
+// Antag stuff down here
+// =====================
+
+/obj/item/mail/traitor
+	var/armed = FALSE
+	var/mob/madeby
+	goodie_count = 0
+
+/obj/item/mail/traitor/envelope
+	name = "envelope"
+	icon_state = "mail_large"
+	stamp_max = 2
+	stamp_offset_y = 5
+
+/obj/item/mail/traitor/attack_self(mob/user)
+	if(recipient_ref)
+		var/datum/mind/recipient = recipient_ref.resolve()
+		// If the recipient's mind has gone, then anyone can open their mail
+		// whether a mind can actually be qdel'd is an exercise for the reader
+		if(recipient && recipient != user?.mind)
+			to_chat(user, span_notice("You can't open somebody else's mail! That's <em>illegal</em>!"))
+			return
+
+	to_chat(user, span_notice("You start to unwrap the package..."))
+	if(!do_after(user, 1.5 SECONDS, target = user))
+		return
+	user.temporarilyRemoveItemFromInventory(src, TRUE)
+	playsound(loc, 'sound/items/poster_ripped.ogg', 50, TRUE)
+	for(var/obj/item/stuff in contents) // Mail and envelope actually can have more than 1 item.
+		if(user.put_in_hands(stuff) && armed)
+			log_bomber(user, "opened armed mail made by [madeby] ([madeby.ckey]), activating", stuff)
+			stuff.attack_self(user)
+	qdel(src)
+
+/obj/item/mail/traitor/attackby(obj/item/W, mob/user, params)
+	. = ..()
```

Comment:
Should we be calling `..()` before checking if it's a multitool? I would guess `..()` shouldn't be called at all for multitools.